failOnWarnings set to false does not appear to function

The following output demonstrates:

```
Run wagoid/commitlint-github-action@v2
  with:
    configFile: ./commitlint.config.js
    firstParent: true
    failOnWarnings: false
    helpURL: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
    token: ***
  env:
    GITHUB_TOKEN: ***
/usr/bin/docker run --name wagoidcommitlintgithubaction200_7879d9 --label 8118cb --workdir /github/workspace --rm -e GITHUB_TOKEN -e INPUT_CONFIGFILE -e INPUT_FIRSTPARENT -e INPUT_FAILONWARNINGS -e INPUT_HELPURL -e INPUT_TOKEN -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/release-it/release-it":"/github/workspace" wagoid/commitlint-github-action:2.0.0
##[error]You have commit messages with errors

⧗   input: chore: bump wagoid/commitlint-github-action from v1.7.0 to v2
⚠   references may not be empty [references-empty]

⚠   found 0 problems, 1 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
```

--- a/src/action.js
+++ b/src/action.js
@@ -104,9 +104,8 @@ const formatErrors = lintedCommits =>
 
 const hasOnlyWarnings = lintedCommits =>
   lintedCommits.length &&
-  lintedCommits.every(
-    ({ lintResult }) => lintResult.valid && lintResult.warnings.length,
-  )
+  lintedCommits.every(({ lintResult }) => lintResult.valid) &&
+  lintedCommits.some(({ lintResult }) => lintResult.warnings.length)
 
 const setFailed = formattedResults => {
   core.setFailed(`You have commit messages with errors\n\n${formattedResults}`)

--- a/src/action.test.js
+++ b/src/action.test.js
@@ -400,12 +400,14 @@ describe('Commit Linter action', () => {
 
     beforeEach(async () => {
       cwd = await git.bootstrap('fixtures/conventional')
+      await gitEmptyCommit(cwd, 'chore: previous commit')
+      await gitEmptyCommit(cwd, 'chore: correct message with no warnings')
       await gitEmptyCommit(
         cwd,
         'chore: correct message\nsome context without leading blank line',
       )
-      const [to] = await getCommitHashes(cwd)
-      await createPushEventPayload(cwd, { to })
+      const [before, from, to] = await getCommitHashes(cwd)
+      await createPushEventPayload(cwd, { before, to })
       updatePushEnvVars(cwd, to)
       td.replace(process, 'cwd', () => cwd)
       td.replace(console, 'log')
@@ -418,6 +420,13 @@ describe('Commit Linter action', () => {
           valid: true,
           errors: [],
           warnings: ['body must have leading blank line'],
+        },
+        {
+          hash: from,
+          message: 'chore: correct message with no warnings',
+          valid: true,
+          errors: [],
+          warnings: [],
         },
       ]
     })

